SAP/spartacus
# fix: allow for resetting to undefined parts of the config of the NgRx Transfer State

fixes https://jira.tools.sap/browse/CXSPA-9122

**QA steps:**
BEFORE:
1. `npm run dev:ssr`
2. Open Chrome [and disable JS](https://developer.chrome.com/docs/devtools/javascript/disable)
3. visit http://localhost:4200/electronics-spa/en/USD/
4. Open Chrome DevTools Elements tab
5. find `<script id="ng-state" type="application/json">` and verify it contains the `cx-state` with the `cms` object
  ![image](https://github.com/user-attachments/assets/183e2dcb-f6c9-4ebd-9501-befae37f8ccf)
  
AFTER:
1. in your app.module provide the following config:
  ```ts
  provideConfig({
    state: {
      ssrTransfer: {
        keys: {
          [cms]: undefined,
        },
      },
    },
  }),
  ```
2. Wait for your app to rebuild WITHOUT TYPESCRIPT ERRORS.
3. Refresh the page in Chrome
4. find `<script id="ng-state" type="application/json">` and verify it contains the `cx-state` WITHOUT the `cms` object 
  ![image](https://github.com/user-attachments/assets/c085ec83-3ecf-4f41-beb0-159f57221edd)
